de_web_plugin.cpp

```diff
@@ -1017,7 +1018,8 @@ void DeRestPluginPrivate::apsdeDataIndication(const deCONZ::ApsDataIndication &i
 
         if (ind.dstAddressMode() == deCONZ::ApsGroupAddress || ind.clusterId() == VENDOR_CLUSTER_ID || ind.clusterId() == IAS_ZONE_CLUSTER_ID ||
             !(zclFrame.frameControl() & deCONZ::ZclFCDirectionServerToClient) ||
-            (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId))
+            (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId) ||
+            (ind.clusterId() == ADEO_CLUSTER_ID && zclFrame.manufacturerCode() == VENDOR_ADEO))
         {
             Sensor *sensorNode = getSensorNodeForAddressAndEndpoint(ind.srcAddress(), ind.srcEndpoint());
```

Comment:
The if statement was already too complex to understand it easily, I'd suggest we split this up a bit:

```cpp
bool checkSensor = false;

if      (ind.dstAddressMode() == deCONZ::ApsGroupAddress) { checkSensor = true; }
else if (ind.clusterId() == VENDOR_CLUSTER_ID) { checkSensor = true; }
else if (ind.clusterId() == IAS_ZONE_CLUSTER_ID) { checkSensor = true; }
else if ((zclFrame.frameControl() & deCONZ::ZclFCDirectionServerToClient) == 0) { checkSensor = true; }
else if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId) { checkSensor = true; }
else if (ind.clusterId() == ADEO_CLUSTER_ID && zclFrame.manufacturerCode() == VENDOR_ADEO) { checkSensor = true; }

if (checkSensor)
{
    ...
}
```

Still a monster but imho easier to reason about compared to one huge if statement. @SwoopX @ebaauw ?

Reply:
@manup I wonder if the ADEO addition is really required. The current code is already broad like a shotgun. @Smanar ?

Making the changes above as suggested would certainly improve readability. We probably should see that we have the optimal order for the checks. Would also love to get the VENDOR ais IAS_ZONE stuff out there any time soon ;)

Reply:
Yeah I m agree, I don't think adeo is a big manufacture, but devices that use their own manufacture cluster can be more.
And it s the same working mode.

Reply:
But how does the Adeo switch send its commands? 

I recon that once of the following might already apply for the incoming commands and Adeo doesn't need to be handled specifically?

```
if      (ind.dstAddressMode() == deCONZ::ApsGroupAddress) { checkSensor = true; }
else if ((zclFrame.frameControl() & deCONZ::ZclFCDirectionServerToClient) == 0) { checkSensor = true; }
else if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId) { checkSensor = true; }
```

Reply:
But like that we will have a problem, I think they use
- cluster command
- client > server

for cluster VENDOR_CLUSTER_ID + IAS_ZONE_CLUSTER_ID + ADEO_CLUSTER_ID

Reply:
But origin client (switch) to server (coordinator) should work?  
That's what `(zclFrame.frameControl() & deCONZ::ZclFCDirectionServerToClient) == 0)`is saying.

Reply:
He yes right, so IDK why the line

`else if ((zclFrame.frameControl() & deCONZ::ZclFCDirectionServerToClient) == 0) { checkSensor = true; }`

Is not enought to make all thoses devices working.

Reply:
Some of the checks can perhaps be removed, but there they were added for some reason at the time, so I guess the then existing checks didn't cut it :)

Reply:
Have found some log

```
18:35:29:884 APS-DATA.indication srcAddr: 0x9257, srcEp: 0x01 dstAddrMode: 2, profile: 0x0104, cluster: 0xFE00, lqi: 255, rssi: -32
18:35:29:885     asdu: 0d771235000d01
18:35:29:885 APS-DATA.indication from child 0x9257
18:35:29:886 Debug adeao 1
18:35:29:887 Debug adeao 2
18:35:29:887 Force binding of attribute reporting for sensor LXEK-5 20
18:35:29:887 [INFO] - No button handler for: LXEK-5, unicast to: 0x0000, endpoint: 0x01, cluster: 0xFE00, command: 0x00, payload: 0D01, zclSeq: 53
```

Still not understand why it s needed.